packages/boba/contracts/contracts/LP/L2LiquidityPool.sol

```diff
@@ -438,6 +450,63 @@ contract L2LiquidityPool is CrossDomainEnabled, ReentrancyGuardUpgradeable, Paus
         }
     }
 
+    /**
+     * Give xBOBA to users who has already deposited liquidity
+     * @param _tokenAddress address of the liquidity token.
+     */
+    function mintXBOBAForPreOwner(
+        address _tokenAddress
+    )
+        internal
+    {
+        address BOBAAddress = Lib_AddressManager(addressManager).getAddress("TK_L2BOBA");
+        if (!xBOBAInfo[msg.sender] && BOBAAddress == _tokenAddress) {
+            // mint xBoba
+            UserInfo storage user = userInfo[_tokenAddress][msg.sender];
+            address xBOBAAddress = Lib_AddressManager(addressManager).getAddress("TK_L2xBOBA");
+            xL2GovernanceERC20(xBOBAAddress).mint(msg.sender, user.amount);
```

Comment:
should we skip calling mint if user.amount is zero, maybe an if "only" for this line?
the next line we need to call anyway every time